jme3-android/src/main/java/com/jme3/app/jmeSurfaceView/JmeSurfaceView.java

```diff
@@ -80,209 +91,411 @@
+     * A state change observer to the holder Activity life cycle, used to keep this android view up-to-date with the holder activity life cycle.
+     * @param source the life cycle source, aka the observable object.
+     * @param event the fired event by the observable object, which is dispatched and sent to the observers.
+     */
+    @Override
+    public void onStateChanged(@NonNull LifecycleOwner source, @NonNull Lifecycle.Event event) {
+        switch (event) {
+            case ON_DESTROY:
+                /*destroy only if the policy flag is enabled*/
+                if(destructionPolicy == DestructionPolicy.DESTROY_WHEN_FINISH) {
+                    destroy();
+                }
+                break;
+            case ON_PAUSE:
+                loseFocus();
+                break;
+            case ON_RESUME:
+                gainFocus();
+                break;
+            case ON_STOP:
+                jmeSurfaceViewLogger.log(Level.INFO, "Context stops, but game is still running");
+                break;
+        }
+    }
+
+    /**
+     * Custom thread that delays the appearance of the display of jme game on the screen for the sake of initial frame pacing and splash screens,
+     * delaying the display of the game (GlSurfaceView) would lead to a substantial delay in the {@link GLSurfaceView.Renderer#onDrawFrame(GL10)} which would
+     * delay invoking both {@link LegacyApplication#initialize()} and {@link LegacyApplication#update()}.
+     * @see JmeSurfaceView#startRenderer(int)
+     * @see OGLESContext#onDrawFrame(GL10 gl)
      */
     private class RendererThread implements Runnable {
         /**
-         * Delays the #{@link GLSurfaceView} add on the UI thread.
+         * Delays the {@link GLSurfaceView} add on the UI thread.
          * @see JmeSurfaceView#NO_DELAY
          * @see JmeSurfaceView#startRenderer(int)
          */
         @Override
         public void run() {
-            /*jme Renderer joins the UIThread at that point*/
-            JmeSurfaceView.this.addView(glSurfaceView);
-            jmeSurfaceViewLogger.log(Level.CONFIG, "JmeSurfaceView's joined the UI thread.......");
+            addGlSurfaceView();
+            jmeSurfaceViewLogger.log(Level.INFO, "JmeSurfaceView's joined the UI thread.......");
         }
     }
 
+    /**
+     * Adds the glSurfaceView to the screen immediately, saving the current app instance.
+     */
+    protected void addGlSurfaceView(){
+        /*jme Renderer joins the UIThread at that point*/
+        JmeSurfaceView.this.addView(glSurfaceView);
+        /*dispatch the layout drawn event*/
+        if(onLayoutDrawn != null){
+            onLayoutDrawn.onLayoutDrawn(legacyApplication, this);
+        }
+        /*set the static memory to hold the game state, only if the game life cycle is not bounded to the activity life cycle*/
```

Comment:
Comment should be changed to `Only if the destruction policy uses KEEP_WHEN_FINISHED policy`
In either polices, the user can choose if he/she wants to keep the surface view updated with a activity life cycle or not !

I will correct this.

Reply:
This comment is wrongly added before the think of adding a destruction policy to control this.